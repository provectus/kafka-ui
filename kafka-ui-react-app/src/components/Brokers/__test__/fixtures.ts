import { BrokersLogdirs } from 'generated-sources';

export const brokersPayload = [
  { id: 1, host: 'b-1.test.kafka.amazonaws.com', port: 9092 },
  { id: 2, host: 'b-2.test.kafka.amazonaws.com', port: 9092 },
];

export const clusterStatsPayload = {
  brokerCount: 2,
  activeControllers: 1,
  onlinePartitionCount: 138,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 239,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [
    { brokerId: 0, segmentSize: 334567, segmentCount: 245 },
    { brokerId: 1, segmentSize: 12345678, segmentCount: 121 },
  ],
  version: '2.2.1',
};

const partition = {
  broker: 2,
  offsetLag: 0,
  partition: 2,
  size: 0,
};
const topics = {
  name: '_confluent-ksql-devquery_CTAS_NUMBER_OF_TESTS_59-Aggregate-Aggregate-Materialize-changelog',
  partitions: [partition],
};

export const brokerLogDirsPayload: BrokersLogdirs[] = [
  {
    error: 'NONE',
    name: '/opt/kafka/data-0/logs',
    topics: [
      {
        ...topics,
        partitions: [partition, partition, partition],
      },
      topics,
      {
        ...topics,
        partitions: [],
      },
    ],
  },
];
