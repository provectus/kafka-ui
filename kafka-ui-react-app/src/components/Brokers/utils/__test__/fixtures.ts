import { BrokerMetrics } from 'generated-sources';

export const brokerMetricsPayload: BrokerMetrics = {
  segmentSize: 23,
  segmentCount: 23,
  metrics: [
    {
      name: 'TotalFetchRequestsPerSec',
      labels: {
        canonicalName:
          'kafka.server:name=TotalFetchRequestsPerSec,topic=_connect_status,type=BrokerTopicMetrics',
      },
      value: 10,
    },
    {
      name: 'ZooKeeperRequestLatencyMs',
      value: 11,
    },
    {
      name: 'RequestHandlerAvgIdlePercent',
    },
  ],
};
export const transformedBrokerMetricsPayload =
  '{"segmentSize":23,"segmentCount":23,"metrics":[{"name":"TotalFetchRequestsPerSec","labels":{"canonicalName":"kafka.server:name=TotalFetchRequestsPerSec,topic=_connect_status,type=BrokerTopicMetrics"},"value":10},{"name":"ZooKeeperRequestLatencyMs","value":11},{"name":"RequestHandlerAvgIdlePercent"}]}';
