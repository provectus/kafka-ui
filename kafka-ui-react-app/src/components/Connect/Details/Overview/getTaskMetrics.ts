import { ConnectorTaskStatus, Task } from 'generated-sources';

export default function getTaskMetrics(tasks?: Task[]) {
  const initialMetrics = {
    running: 0,
    failed: 0,
  };

  if (!tasks) {
    return initialMetrics;
  }

  return tasks.reduce((acc, { status }) => {
    const state = status?.state;
    if (state === ConnectorTaskStatus.RUNNING) {
      return { ...acc, running: acc.running + 1 };
    }
    if (state === ConnectorTaskStatus.FAILED) {
      return { ...acc, failed: acc.failed + 1 };
    }
    return acc;
  }, initialMetrics);
}
