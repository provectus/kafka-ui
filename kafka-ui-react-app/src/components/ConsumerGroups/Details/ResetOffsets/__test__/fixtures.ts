export const expectedOutputs = {
  EARLIEST: {
    partitions: [0],
    partitionsOffsets: [
      {
        offset: undefined,
        partition: 0,
      },
    ],
    resetType: 'EARLIEST',
    topic: '__amazon_msk_canary',
  },
  LATEST: {
    partitions: [0],
    partitionsOffsets: [
      {
        offset: undefined,
        partition: 0,
      },
    ],
    resetType: 'LATEST',
    topic: '__amazon_msk_canary',
  },
  OFFSET: {
    partitions: [0],
    partitionsOffsets: [
      {
        offset: '10',
        partition: 0,
      },
    ],
    resetType: 'OFFSET',
    topic: '__amazon_msk_canary',
  },
};
