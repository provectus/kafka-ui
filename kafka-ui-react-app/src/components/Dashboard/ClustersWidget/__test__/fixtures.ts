import { Cluster, ServerStatus } from 'generated-sources';

export const onlineCluster: Cluster = {
  name: 'secondLocal',
  defaultCluster: false,
  status: ServerStatus.Online,
  brokerCount: 1,
  onlinePartitionCount: 6,
  topicCount: 3,
  bytesInPerSec: 0.000030618196853764715,
  bytesOutPerSec: 5.737800890036267075817,
};

export const offlineCluster: Cluster = {
  name: 'local',
  defaultCluster: true,
  status: ServerStatus.Offline,
  brokerCount: 1,
  onlinePartitionCount: 2,
  topicCount: 2,
  bytesInPerSec: 8000.0000067376808542600021,
  bytesOutPerSec: 0.8153063567297119490871,
};

export const clusters: Cluster[] = [onlineCluster, offlineCluster];
