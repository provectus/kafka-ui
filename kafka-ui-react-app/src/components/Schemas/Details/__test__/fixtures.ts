import { SchemaSubject, SchemaType } from 'generated-sources';
import {
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
} from 'redux/reducers/schemas/__test__/fixtures';

export const versionPayload = [
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
];
export const versionEmptyPayload = [];

export const jsonSchema: SchemaSubject = {
  subject: 'test',
  version: '15',
  id: 1,
  schema:
    '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
  compatibilityLevel: 'BACKWARD',
  schemaType: SchemaType.JSON,
};

export const protoSchema: SchemaSubject = {
  subject: 'test_proto',
  version: '1',
  id: 2,
  schema:
    'syntax = "proto3";\npackage com.indeed;\n\nmessage MyRecord {\n  int32 id = 1;\n  string name = 2;\n}\n',
  compatibilityLevel: 'BACKWARD',
  schemaType: SchemaType.PROTOBUF,
};
