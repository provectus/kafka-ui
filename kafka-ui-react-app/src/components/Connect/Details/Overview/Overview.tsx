import React from 'react';
import * as C from 'components/common/Tag/Tag.styled';
import * as Metrics from 'components/common/Metrics';
import getTagColor from 'components/common/Tag/getTagColor';
import { RouterParamsClusterConnectConnector } from 'lib/paths';
import useAppParams from 'lib/hooks/useAppParams';
import { useConnector, useConnectorTasks } from 'lib/hooks/api/kafkaConnect';

import getTaskMetrics from './getTaskMetrics';

const Overview: React.FC = () => {
  const routerProps = useAppParams<RouterParamsClusterConnectConnector>();

  const { data: connector } = useConnector(routerProps);
  const { data: tasks } = useConnectorTasks(routerProps);

  if (!connector) {
    return null;
  }

  const { running, failed } = getTaskMetrics(tasks);

  return (
    <Metrics.Wrapper>
      <Metrics.Section>
        {connector.status?.workerId && (
          <Metrics.Indicator label="Worker">
            {connector.status.workerId}
          </Metrics.Indicator>
        )}
        <Metrics.Indicator label="Type">{connector.type}</Metrics.Indicator>
        {connector.config['connector.class'] && (
          <Metrics.Indicator label="Class">
            {connector.config['connector.class']}
          </Metrics.Indicator>
        )}
        <Metrics.Indicator label="State">
          <C.Tag color={getTagColor(connector.status.state)}>
            {connector.status.state}
          </C.Tag>
        </Metrics.Indicator>
        <Metrics.Indicator label="Tasks Running">{running}</Metrics.Indicator>
        <Metrics.Indicator
          label="Tasks Failed"
          isAlert
          alertType={failed > 0 ? 'error' : 'success'}
        >
          {failed}
        </Metrics.Indicator>
      </Metrics.Section>
    </Metrics.Wrapper>
  );
};

export default Overview;
