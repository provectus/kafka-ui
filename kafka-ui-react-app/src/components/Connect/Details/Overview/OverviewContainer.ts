import { connect } from 'react-redux';
import { RootState } from 'redux/interfaces';
import {
  getConnector,
  getConnectorRunningTasksCount,
  getConnectorFailedTasksCount,
} from 'redux/reducers/connect/selectors';

import Overview from './Overview';

const mapStateToProps = (state: RootState) => ({
  connector: getConnector(state),
  runningTasksCount: getConnectorRunningTasksCount(state),
  failedTasksCount: getConnectorFailedTasksCount(state),
});

export default connect(mapStateToProps)(Overview);
