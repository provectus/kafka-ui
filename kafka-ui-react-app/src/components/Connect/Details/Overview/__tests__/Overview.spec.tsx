import React from 'react';
import Overview from 'components/Connect/Details/Overview/Overview';
import { connector, tasks } from 'lib/fixtures/kafkaConnect';
import { screen } from '@testing-library/react';
import { render } from 'lib/testHelpers';
import { useConnector, useConnectorTasks } from 'lib/hooks/api/kafkaConnect';

jest.mock('lib/hooks/api/kafkaConnect', () => ({
  useConnector: jest.fn(),
  useConnectorTasks: jest.fn(),
}));

describe('Overview', () => {
  it('is empty when no connector', () => {
    (useConnector as jest.Mock).mockImplementation(() => ({
      data: undefined,
    }));
    (useConnectorTasks as jest.Mock).mockImplementation(() => ({
      data: undefined,
    }));

    render(<Overview />);
    expect(screen.queryByText('Worker')).not.toBeInTheDocument();
  });

  describe('when connector is loaded', () => {
    beforeEach(() => {
      (useConnector as jest.Mock).mockImplementation(() => ({
        data: connector,
      }));
    });
    beforeEach(() => {
      (useConnectorTasks as jest.Mock).mockImplementation(() => ({
        data: tasks,
      }));
    });

    it('renders metrics', () => {
      render(<Overview />);

      expect(screen.getByText('Worker')).toBeInTheDocument();
      expect(
        screen.getByText(connector.status.workerId as string)
      ).toBeInTheDocument();

      expect(screen.getByText('Type')).toBeInTheDocument();
      expect(
        screen.getByText(connector.config['connector.class'] as string)
      ).toBeInTheDocument();

      expect(screen.getByText('Tasks Running')).toBeInTheDocument();
      expect(screen.getByText(2)).toBeInTheDocument();
      expect(screen.getByText('Tasks Failed')).toBeInTheDocument();
      expect(screen.getByText(1)).toBeInTheDocument();
    });
  });
});
