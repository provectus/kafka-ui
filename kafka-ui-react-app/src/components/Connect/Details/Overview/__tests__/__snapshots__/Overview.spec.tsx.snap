// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview view matches snapshot 1`] = `
.c5 {
  border: none;
  border-radius: 16px;
  height: 20px;
  line-height: 20px;
  background-color: #FFEECC;
  color: #171A1C;
  font-size: 12px;
  display: inline-block;
  padding-left: 0.75em;
  padding-right: 0.75em;
  text-align: center;
}

.c0 {
  padding: 1.5rem 1rem;
  background: #F1F2F3;
  margin-bottom: 0.5rem !important;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c3 {
  background-color: #FFFFFF;
  height: 68px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 150px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 12px 16px;
  box-shadow: 3px 3px 3px rgba(0,0,0,0.08);
  margin: 0 0 3px 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  font-weight: 500;
  font-size: 12px;
  color: #73848C;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 > .c2:first-child {
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
}

.c1 > .c2:last-child {
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
}

.c6 {
  grid-area: status;
  fill: none;
  width: 4px;
  height: 4px;
}

.c7 {
  fill: #E51A1A;
}

@media screen and (max-width:1023px) {
  .c1 > .c2:first-child,
  .c1 > .c2:last-child {
    border-radius: 0;
  }
}

<div
  className="c0"
>
  <div>
    <div
      className="c1"
    >
      <div
        className="c2 c3"
      >
        <div>
          <div
            className="c4"
          >
            Worker
             
          </div>
          <span>
            kafka-connect0:8083
          </span>
        </div>
      </div>
      <div
        className="c2 c3"
      >
        <div>
          <div
            className="c4"
          >
            Type
             
          </div>
          <span>
            SOURCE
          </span>
        </div>
      </div>
      <div
        className="c2 c3"
      >
        <div>
          <div
            className="c4"
          >
            Class
             
          </div>
          <span>
            FileStreamSource
          </span>
        </div>
      </div>
      <div
        className="c2 c3"
      >
        <div>
          <div
            className="c4"
          >
            State
             
          </div>
          <span>
            <p
              className="c5"
              color="yellow"
            >
              RUNNING
            </p>
          </span>
        </div>
      </div>
      <div
        className="c2 c3"
      >
        <div>
          <div
            className="c4"
          >
            Tasks Running
             
          </div>
          <span>
            10
          </span>
        </div>
      </div>
      <div
        className="c2 c3"
      >
        <div>
          <div
            className="c4"
          >
            Tasks Failed
             
            <svg
              className="c6"
              viewBox="0 0 4 4"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                className="c7"
                cx={2}
                cy={2}
                r={2}
              />
            </svg>
          </div>
          <span>
            f
            2
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
